bookwyrm/models/readthrough.py

```diff
@@ -24,3 +31,26 @@ def save(self, *args, **kwargs):
         self.user.last_active_date = timezone.now()
         self.user.save()
         super().save(*args, **kwargs)
+
+    def create_update(self):
+        if self.progress:
+            return self.progressupdate_set.create(
+                user=self.user,
+                progress=self.progress,
+                mode=self.progress_mode)
+
+class ProgressUpdate(BookWyrmModel):
+    ''' Store progress through a book in the database. '''
+    user = models.ForeignKey('User', on_delete=models.PROTECT)
+    readthrough = models.ForeignKey('ReadThrough', on_delete=models.PROTECT)
+    progress = models.IntegerField()
```

Comment:
I think this should have a min value validator to prevent negative values here

Reply:
I added validators here and on ReadThrough.progress (and tests to make sure they work), but didn't actually use them anywhere. I'm not sure they'll have any effect on ProgressUpdate, but might on the ReadThrough depending on how we use them?

In any case, having the validators is better than not